### ISSUE TITLE
Bug: `TypeError` in backward pass when `bias=None`

### ISSUE THREAD

**Issue by zhiyuan1i** [Dao-AILab/causal-conv1d#61]
**Environment:**
* **PyTorch Version:** 2.7.1
* **causal_conv1d Version:** 1.5.0.post8

**Description:**

A `TypeError` occurs during the backward pass of `causal_conv1d_fn` when it is called without a bias tensor (`bias=None`). The forward pass completes without issues, but the gradient computation fails because the implementation does not handle the `None` case for the bias gradient.

This issue is reliably reproduced by the test suite when `has_bias` is set to `False`.

**Reproduction Script:**

The following script, adapted from the test suite, demonstrates the failure. The error is triggered when `has_bias` is `False`.

```python
import torch
from einops import rearrange
from causal_conv1d.causal_conv1d_interface import causal_conv1d_fn

# Parameters to trigger the bug
B, T, D, W = 4, 500, 128, 4
activation = 'swish'
has_bias = False # This triggers the bug
has_residual = False
dtype = torch.float16
device = "cuda" if torch.cuda.is_available() else "cpu"

print("--- Causal Conv1d Backward Pass Test ---")
print(f"Parameters: B={B}, T={T}, D={D}, W={W}, activation={activation}, has_bias={has_bias}, dtype={dtype}")

torch.manual_seed(42)

x = torch.randn(B, T, D, device=device, dtype=dtype).requires_grad_(True)
weight = torch.randn(D, W, device=device, dtype=dtype).requires_grad_(True)
bias = torch.randn(D, device=device, dtype=dtype).requires_grad_(True) if has_bias else None
residual = x.detach().clone().requires_grad_(True) if has_residual else None
dy = torch.randn(B, T, D, device=device, dtype=dtype)

print(f"Bias is None: {bias is None}")

try:
    # Forward pass
    ref = causal_conv1d_fn(
        x=rearrange(x, "b t d -> b d t"),
        weight=weight,
        bias=bias,
        activation=activation,
    )
    ref = rearrange(ref, "b d t -> b t d")
    if has_residual:
        ref += residual

    # Backward pass
    print("Executing backward pass...")
    ref.backward(dy)
    print("Backward pass successful.")

except TypeError as e:
    print("\n!!! TEST FAILED !!!")
    print("Caught expected TypeError during backward pass.")
    print(f"Error: {e}")
    print("\nTraceback:")
    import traceback
    traceback.print_exc()
```
Root Cause:

The error is located at: https://github.com/Dao-AILab/causal-conv1d/blob/main/causal_conv1d/cpp_functions.py#L135

The line dbias = torch.zeros_like(bias, dtype=torch.float32) is executed without first checking if bias is a Tensor. When bias is None, this call raises a TypeError. A subsequent call to dbias.type_as(bias) would also fail.
